### ISSUE TITLE
Previous user’s data shown to Lurker after user logs out - bundles and search results

### ISSUE THREAD

**Issue by slobulike** [riverscuomo/public-bug-hunt#82]
**Describe the bug**
A bundle purchased by the previous user is shown on the lurker profile screen after logging out from the previous user's account.

**To Reproduce**
1. Create a new user.
2. Purchase a bundle.
3. Log out.
4. Click "Maybe later" on the login screen
5. Browse to the profile screen
6. The bundle is shown against the lurker profile 

**Expected behavior**
No data from the previous user should be shown.

**Screenshots**
![Screenshot_20231027-211348](https://github.com/riverscuomo/public-bug-hunt/assets/48949712/ed04c943-2ccd-4f40-957e-add428c307d9)

**Smartphone (please complete the following information):**
 - Device: [Pixel 6a]
 - OS: Android 14
 - Browser [App]
 - Version [183 v1.10.2]

**Additional context**

**Comment by SlightlyCompletely** [comment]
Also occurs on iPad.

Additionally, if a song from the purchased playlist is playing when user logs out of their account, the song will keep playing even if entering back into Weezify as a lurker without having the bundle purchased. The lurker cannot restart the unowned track after stopping it though. 

A lurker visiting the search page will also see the previous user’s last search results, if search was used.

Device: iPad Air 5th gen
OS: iPadOS 17.0.3
Weezify 1.12.0 (186)

**Comment by slobulike** [comment]
In Android version 1.13.1 (189) this has extended further so that until the app is closed a newly created user is able to play all tracks purchased by the previous logged in user. 

Essentially the "Your library" page users the previous users data. Maybe the player bloc refactor means that cache isn't clearing? 

When I attempted to clear the cache using the new button, I discovered that whilst in this state of having two users data loaded (profile shows correct logged in user data, Your Library shows previous users data) the profile page does not load correctly and the logout, delete, and clear cache buttons don't show e.g.

![Screenshot_20231213-123253](https://github.com/riverscuomo/public-bug-hunt/assets/48949712/f95c6c13-ebdf-45b7-8a26-5154b7d6037a)
